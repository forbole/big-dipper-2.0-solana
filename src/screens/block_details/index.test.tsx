/* eslint-disable */
import React from 'react';
import { createMockClient } from 'mock-apollo-client';
import { ApolloProvider } from '@apollo/client';
import renderer from 'react-test-renderer';
import {
  MockTheme, wait,
} from '@tests/utils';
import {
BlockDetailsDocument,
} from '@graphql/types';
import BlockDetails from '.';

// ==================================
// mocks
// ==================================
jest.mock('next/router', () => ({
  useRouter: () => ({
    query: {
      height: 300,
    },
  }),
}));

jest.mock('@components', () => ({
  Layout: (props) => <div id="Layout" {...props} />,
  LoadAndExist: (props) => <div id="LoadAndExist" {...props} />,
}));

jest.mock('./components', () => ({
  Overview: (props) => <div id="Overview" {...props} />,
  Transactions: (props) => <div id="Transactions" {...props} />,
  Signatures: (props) => <div id="Signatures" {...props} />,
}));

const mockAverageBlockTime = jest.fn().mockResolvedValue({
  "data": {
    "block": [
      {
        "slot": 125042260,
        "hash": "6G2U5iuU56dZS8nmHvv7n5RgYm8wTEdDm1srx5ZGdHy5",
        "leader": "7Gjec4iDbTxLvVYNsRbZrrHdtyLByzdDJ1C5BmcMMBks",
        "timestamp": "2022-03-15T10:55:19",
        "numTxs": 713
      }
    ],
    "transaction": [
      {
        "slot": 125042260,
        "hash": "3FZ3GH7QcwvcQgdF8r6Ap6UHMgY49ga4aYfFzy2nEADKajLLzzYhUwvcvdLqMeYQaAYiDf3JnJpyVRYoMCb1Zv7u",
        "error": true,
        "block": {
          "timestamp": "2022-03-15T10:55:19"
        },
        "numInstructions": 1
      }
    ]
  }
});

// ==================================
// unit tests
// ==================================
describe('screen: BlockDetails', () => {
  it('matches snapshot', async () => {
    const mockClient = createMockClient();
    mockClient.setRequestHandler(
      BlockDetailsDocument,
      mockAverageBlockTime,
    );

    let component;
    renderer.act(() => {
      component = renderer.create(
        <ApolloProvider client={mockClient}>
          <MockTheme>
            <BlockDetails />
          </MockTheme>
        </ApolloProvider>,
      );
    });
    await wait();

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
