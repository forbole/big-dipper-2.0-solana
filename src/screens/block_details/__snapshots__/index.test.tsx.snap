// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: BlockDetails matches snapshot 1`] = `
<div
  id="Layout"
  navTitle="blocks:blockDetails"
>
  <div
    exists={true}
    id="LoadAndExist"
    loading={false}
  >
    <span
      className="makeStyles-root"
    >
      <div
        hash="4YJBguXbyjWNQ6o9SCCmUWab9KbxgAkmKdopCikRNSd7"
        id="Overview"
        leader="9QU2QSxhb24FUX3Tu2FpczXjpK3VYrvRudywSZaM29mF"
        slot={125048761}
        timestamp="2022-03-15T11:57:26"
        txs={1421}
      />
      <div
        id="Transactions"
        transactions={
          Array [
            Object {
              "numInstructions": 1,
              "signature": "AxVwePxYYKqGHSwrrpjwptp9km4N91fYtQcHgYvLNRKPEMp6ZNdQJd6wxA4KF3LHRbKS7iZwMWckmnT9YdATE2Y",
              "slot": 125056600,
              "success": true,
              "timestamp": "2022-03-15T11:57:26",
            },
          ]
        }
      />
    </span>
  </div>
</div>
`;
